pkg/transparentproxy/transparentproxy.go

```diff
@@ -0,0 +1,18 @@
+package transparentproxy
+
+import (
+	"github.com/kumahq/kuma/pkg/transparentproxy/config"
+	"github.com/kumahq/kuma/pkg/transparentproxy/istio"
+)
+
+type TransparentProxy interface {
+	// returns the stdout and stderr as string and error if such has occurred
+	Setup(cfg *config.TransparentProxyConfig) (string, error)
+
+	// returns the stdout and stderr as string and error if such has occurred
+	Cleanup(dryRun bool) (string, error)
+}
+
+func GetDefaultTransparentProxy() TransparentProxy {
```

Comment:
nit: just `DefaultTransparentProxy()` without `Get`, to follow our convention in `components.go` files.